text/0000-light-dom-scoped-styles.md

```diff
@@ -0,0 +1,330 @@
+
+However, our goal here is not to maintain synthetic shadow DOM for all eternity. So if light DOM scoped styles differ from native shadow DOM in favor of simplicity, then this will be better in the long run for performance and maintainability.
+
+## Alternatives
+
+### Not scoping
+
+We could simply not implement scoped CSS for light DOM. However, given how popular it is in other frameworks (Svelte, Vue, the wide ecosystem of React CSS-in-JS libraries), it seems like a shame for LWC not to support it.
+
+### Descendant selectors
+
+Another alternative is to use CSS selectors that can style all of a component's descendants. For instance:
+
+```css
+/* input */
+div {}
+```
+
+```css
+/* output */
+.scope-token div {}
+```
+
+This approach was rejected because, although it's similar to the style scoping used in Aura, it's dissimilar from the style scoping used in native shadow DOM, Vue, Svelte, Stencil, `styled-components`, etc. Our hope is that the current approach will be more familiar to more developers.
+
+If developers want a component to contain styles that affect its children, it's always possible to use non-scoped light DOM styles, and to target the child component's classes, attributes, etc.:
+
+```css
+/* parent.css */
+.my-child {}
+```
+
+```html
+<!-- child.html -->
+<template lwc:render-mode="light">
+  <div class="my-child"></div>
+</template>
+```
+
+This is the same solution one might use with global styles in Vue (`<style>`) or Svelte (`:global()`).
+
+### Targeting the root element with `x-component`
+
+In theory, a light DOM component can target its own root element in both global and scoped styles by using its own tag name, e.g. `x-component`:
+
+```css
+x-component {
+    background: red;
+}
+```
+
+The problem with this approach is that templates do not really "own" their tag names. A subclass of a `LightningElement`, for instance, could end up with a totally different tag name. Or we may rename tags at the platform level. Therefore it's not safe for developers to use this technique to "guess" their own tag name.
+
+This is why we support the `:host` selector in scoped CSS instead.
+
+## Adoption strategy
+
+Light DOM components are already opt-in (using `static renderMode = 'light'`), and scoped light DOM styles would also be opt-in, using `*.scoped.css`.
+
+# How we teach this
+
+Conceptually, scoped light DOM styles will have to be bundled up into a larger discussion of light DOM versus shadow DOM. Because switching between the two will never be as simple as flipping a boolean flag, and because the differences between the two can have a wide-ranging impact, we have to be careful about how we communicate the differences to developers.
+
+# Unresolved questions
+
+There are no unresolved questions at this time.

```

Comment:
To add: should `*.css` be the global, or should it be the scoped one, and `*.global.css` is the global one? In Vue, global is the default, whereas in Svelte, scoped is the default. There are different intuitions about this, so it's not clear one or the other is the best.

Reply:
Is there any support for `slot` in this world?

Reply:
Anything different about custom properties? 

Thinking of this extreme case, what would the color of the slotted `span` be? `cornflowerblue` or `blue`?

```html
<x-app>
  <style>
    :root {
      --color-red: red;
      --color-blue: blue;
    }
  </style>
  <style class="x-foo_foo" type="text/css">
    div.x-foo_foo { 
      --color-red: maroon; 
      color: var(--color-blue); 
    }
    div.x-foo_foo span {
      --color-blue: cornflowerblue; 
    }
  </style>
  
  <div class="x-foo_foo">
    <h2>Hello</h2>
    <x-bar>
      #shadowRoot
        <style>
          h2 {
            color: var(--color-red);
          }
          ::slotted(span) {
            color: var(--color-blue)
          }
        </style>
        <h2>Hello</h2>
        <slot>↦ span</slot>
      #lightdom
      <span>slotted element</span>
    </x-bar>
  </div>
</x-app>
```

Reply:
This was a bit of a brain-teaser, so I wrote [a demo](https://github.com/nolanlawson/lwc-barebone/commit/167c105edb0a20ed89b0152426966099144de184). 🙂 

Looks like the answer is cornflower blue. It's a bit confusing because both `color` and custom properties are inherited, but basically yes, the `--color-blue` inherits down from the `<div>`.

![Screen Shot 2021-09-30 at 12 47 48 PM](https://user-images.githubusercontent.com/283842/135521008-b25fa6aa-6b65-4fef-8af2-a9304b3c502c.png)

(Using Firefox but it's the same in Chrome; just easier to see in Firefox since Chrome doesn't visualize constructable stylesheets very well.)